docs/develop/howtos/override_components.md

```diff
@@ -0,0 +1,121 @@
+│   │   |   |   ├── invenio_app_rdm
+│   │   |   |   |   ├── deposit
+│   │   |   |   |   |   ├── override.js
+│   │   |   |   |   |   ├── RDMDepositForm.js
+```
+
+We will override the file `override.js`.
+
+- Find a react component to override. For this example, we are going to override the deposit form '`Metadata-only record`' checkbox.
+
+!['`Metadata-only record`' checkbox](./img/metadata_only_checkbox.png)
+
+!!! info "You must know the component id beforehand"
+    In order to override a component, the exact overridable id must be retrieved beforehand.
+    For now, you must retrieve the component id directly from the module's code.
+
+The react component's overridable id for the '`Metadata-only record`' checkbox is `ReactInvenioDeposit.MetadataAccess.layout`.
+
+For this example, we will replace the '`Metadata-only record`' checkbox with another component, like a [Toggle](https://react.semantic-ui.com/modules/checkbox/#types-toggle) component.
+
+- Create a file named `override.js` with the same path as the one defined by the module (e.g. `<instance_name>/assets/js/invenio_app_rdm/deposit/override.js`). Note that `semantic-ui` was ommited.
+
+- Create a file (e.g. `MetadataToggle.js`) to implement your component, under `<instance_name>/assets/js/components`. If the folder `components` does not exist, you can create it.
+
+Your instance's assets folder should look like this by now:
+
+```terminal
+├── <instance_name>
+│   ├── assets
+│   │   ├── js
+│   │   |   ├── invenio_app_rdm
+│   │   |   |   ├── deposit
+│   │   |   |   |   ├── override.js
+│   │   |   |   |   ├── RDMDepositForm.js
+│   │   |   ├── components
+│   │   |   |   ├── MetadataToggle.js
+```
+
+- Create a react component to override the default one. In the file `MetadataToggle.js`, add the following code:
+
+    ```javascript
+    import React from "react";
+    import { Checkbox } from "semantic-ui-react";
+    import { useFormikContext } from "formik";
+    import PropTypes from "prop-types";
+
+    const MetadataToggle = (props) => {
+    const { filesEnabled } = props;
+    const { setFieldValue } = useFormikContext();
+
+    const handleOnChangeMetadataOnly = () => {
+        setFieldValue("files.enabled", !filesEnabled);
+        setFieldValue("access.files", "public");
+    };
+
+    return (
+        <Checkbox
+        toggle
+        label="Metadata-only record"
+        onChange={handleOnChangeMetadataOnly}
+        />
+    );
+    };
+
+    export default MetadataToggle;
+
+    MetadataToggle.propTypes = {
+    filesEnabled: PropTypes.bool.isRequired,
+    };
+    ```
+
+- Add the following object to `override.js`, specifying both components: the one to be replaced and its replacement.
+
+```javascript
+import MetadataToggle from "../../components/MetadataToggle";
+
+export const overriddenComponents = {
+  "ReactInvenioDeposit.MetadataOnlyToggle.layout": MetadataToggle,
+};
+```
+
+- Lastly, rebuild your assets. In a terminal, run the following command:
```

Comment:
```suggestion
- Lastly, rebuild your assets - the new file needs to be added to your instance. In a terminal, run the following command:
```
maybe these small justification would help people to learn on which command should be run when